### PR TITLE
Alternative fix for qsim and cirq opposite ordering of qubits

### PR DESCRIPTION
**Pull request by gecrooks**
Rather than transposing the amplitudes (which is slow), instead number qubits in reverse order when constructing the qsim circuit.

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
Also - how does this not affect `qsimcirq_test`? That seems to suggest that at least one input route is not being reversed correctly.

File: qsimcirq/qsim_simulator.py [hunk 1]
```diff
@@ -37,12 +37,9 @@ def __init__(self,
                  qsim_data: np.ndarray,
                  qubit_map: Dict[ops.Qid, int]):
       # Generate state vector from qsim results
-      amplitudes = qsim_data
-      N = len(qubit_map)
-      amplitudes = np.reshape(amplitudes, [2]*(N+1))
-      amplitudes = np.transpose(amplitudes)
-      amplitudes = amplitudes[0] + 1j * amplitudes[1]
-      state_vector = amplitudes.flatten()
+      state_vector = np.reshape(qsim_data, [-1, 2])
```
[the remark this answers is not in the document]
Comment (gecrooks) [reply qsimcirq/qsim_simulator.py]: Right, it just needs to do the float->complex conversion.

**Comment by gecrooks** [comment]
That's a good point. How come I didn't have to change qsimh_simulator.py also?

File: qsimcirq/qsim_circuit.py [hunk 2]
```diff
@@ -60,6 +60,10 @@ def translate_cirq_to_qsim(
     circuit_data = []
     ordered_qubits = cirq.ops.QubitOrder.as_qubit_order(qubit_order).order_for(
         self.all_qubits())
+
+    # qsim numbers qubits in reverse order from cirq
+    ordered_qubits = list(reversed(ordered_qubits))
```
Comment (95-martin-orion) [inline qsimcirq/qsim_circuit.py]: With this line commented out, the full-state test still passes. The culprit appears to be these lines:
https://github.com/quantumlib/qsim/blob/88f187234645c92a8a9bc77ad6f425b24644dcb0/qsimcirq/qsim_simulator.py#L151-L153

Note that `ordered_qubits` is (incorrectly) defined using the pre-translation version of the circuit.

**Pull request merged**